cmd/gox/template/project/.builds/web/worker/go-wasm-loader.js

```diff
@@ -0,0 +1,219 @@
+      }, 10000); // 10秒超时
+    });
+  };
+}
+
+function tryRunGoWasm() {
+  const workerId = (typeof Module !== 'undefined' && Module['workerID']) || 'unknown';
+  if (!Module["FFI"]) {
+    return;
+  }
+  if (!Module["gameProjectData"]) {
+    return;
+  }
+
+  if (self.goBridge && self.goBridge.isReady) {
+    try {
+      // If Go WASM is ready, can call related functions to process data
+      self.goBridge.callGoFunctionSafe('goLoadData', Module["gameProjectData"]);
+      callMainThread('onGameStarted');
+    } catch (error) {
+      console.error(`[Worker ${workerId}] Error calling Go function to process project data:`, error);
+    }
+  }
+}
+
+/**
+ * Initializes Go WASM on demand (callable from any thread)
+ * This function will be called on godot_js_spx_on_engine_start callback
+ */
+async function initExtensionWasm() {
+  var createWrapper = function (module, name) {
+    return function () {
+      return module["asm"][name].apply(null, arguments);
+    };
+  }
+  const workerId = Module['workerID'] || 'main';
+  const threadInfo = typeof importScripts !== 'undefined' ? 'Worker' : 'MainThread';
+  Module._cmalloc = createWrapper(Module, "malloc");
+  Module._cfree = createWrapper(Module, "free");
+  FFI = null
+
+  try {
+    // Load Go WASM module
+    await loadGoWasmModule();
+    FFI = Module["FFI"];
+    self.tryRunGoWasm()
+    return true;
+  } catch (error) {
+    console.error(`[Thread ${threadInfo}-${workerId}] Go WASM initialization failed:`, error);
+    return false;
+  }
+}
+
+// AI Token Provider related variables and functions
+let tokenRequestId = 0;
+const pendingTokenRequests = new Map();
+
+// requestTokenFromMainThread function is no longer needed, because now using the generic function proxy mechanism
+
+function handleCallResponse(data) {
+  if (data.responseId) {
+    const requestId = parseInt(data.responseId);
+    if (pendingTokenRequests.has(requestId)) {
+      const { resolve, reject } = pendingTokenRequests.get(requestId);
+      pendingTokenRequests.delete(requestId);
+
+      if (data.error) {
+        reject(new Error(data.error));
+      } else {
+        resolve(data.result || "");
+      }
+      return;
+    }
+    console.error("handleCallResponse: no pendingTokenRequests", data)
+  }
+  console.error("handleCallResponse: no responseId", data)
+}
+
+// Expose functions to global scope for godot.editor.js to call
+if (typeof self !== 'undefined') {
+  self.initExtensionWasm = initExtensionWasm;
+}
+
+
+/**
+ * Simplified Go WASM module loading logic
+ */
+async function loadGoWasmModule() {
+  // If already loaded, return immediately
+  if (self.goBridge && self.goBridge.isReady) {
+    console.log(`[Godot Worker ${Module['workerID']}] Go WASM is already loaded, using directly`);
+    return;
+  }
+
+  try {
+    // Create Go WASM Bridge instance
+    const goBridge = new GoWasmBridge();
+
+    // Initialize Go WASM module
+    await goBridge.initialize({
+      wasmPath: './gdspx.wasm',
```

Comment:
Is it possible for me (XBuilder, or other projects that depends on spx runner) to specify the wasm URL now?

And is it possible to specify the URL for scripts like `go-wasm-loader.js`? Seems that they are loaded dynamically, too.